### PR TITLE
feat: add rxjs operators `poll` and `whenDocumentVisible`

### PR DESCRIPTION
**Pull request by FabienDehopre**
Add two RxJs operators that works well together but still can be used independently.

`poll` can be used to poll an API at a fixed interval with an optional initial delay.
`whenDocumentVisible` is the RxJs counterpart of the `injectDocumentVisibility` and can be used to pause and resume a RxJs stream based on the document visibility state.

Related to #405

### CODE REVIEW

**Comment by eneajaho** [comment]
PR not found

**Comment by FabienDehopre** [comment]
What do you mean?

**Review by nartc — APPROVED** [review]

**Comment by nartc** [comment]
@all-contributors please add @FabienDehopre for code

File: docs/src/content/contributors/fabiendehopre.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+	"name": "Fabien Dehopré",
+	"twitter": "https://twitter.com/FabienDehopre",
+	"github": "https://github.com/FabienDehopre",
+	"linkedin": "https://www.linkedin.com/in/fabien1979/?lipi=urn%3Ali%3Apage%3Ad_flagship3_feed%3BbRilI4BMRLygBScKCu5i%2Fw%3D%3D"
```
Comment (ajitzero) [inline docs/src/content/contributors/fabiendehopre.json:5]: minor: 😅 
```suggestion
	"linkedin": "https://www.linkedin.com/in/fabien1979/"
```

File: libs/ngxtension/poll/src/poll.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+export function poll<T>(
+	period: number,
+	initialDelay = 0,
+): MonoTypeOperatorFunction<T> {
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+	return (source: Observable<T>) => {
+		return timer(initialDelay, period).pipe(
+			concatMap(() => {
+				return source;
+			}),
```
Comment (ajitzero) [inline libs/ngxtension/poll/src/poll.ts:24]: nit: formatting
```suggestion
			concatMap(() => source),
```

**Review by ajitzero — APPROVED** [review]

File: libs/ngxtension/poll/src/poll.ts (hunk 2 above)
Comment (FabienDehopre) [reply libs/ngxtension/poll/src/poll.ts]: fixed

**Comment by nartc** [comment]
https://github.com/all-contributors please add @FabienDehopre for code

**Comment by nartc** [comment]
@FabienDehopre looks like the test failed. Can you take a look?

**Comment by nartc** [comment]
@all-contributors please add @FabienDehopre for code

(third time the charm?)

**Comment by nartc** [comment]
@all-contributors please add @FabienDehopre for code

(fourth time 🥲)

**Comment by FabienDehopre** [comment]
I've fixed the failing tests.

**Pull request merged**